### PR TITLE
[SQL] Explain that WATERMARK is evaluated in the next circuit step

### PR DESCRIPTION
**Pull request by mihaibudiu**
I have verified manually that this is true using the web ui.

### CODE REVIEW

File: docs/sql/streaming.md [hunk 1]
```diff
@@ -121,6 +121,10 @@ watermark.  Given a `WATERMARK` annotation with value W, an input row
 with a value X for a watermarked column will be "held up" until an
 input row with a value X + W has been received.  The program will
 behave as if the row with value X has only just been received.
+Similar to `LATENESS`, the effect of the `WATERMARK` is visible in the
+*next* circuit step; an event that changes the watermark in step 2 of
+the circuit will only make "visible" the eligible events inserted up
+to step 1.
```
Comment (lalithsuresh) [inline docs/sql/streaming.md:128]: Do we document what a step is anywhere?

Comment (mihaibudiu) [reply docs/sql/streaming.md:128]: in the dbsp paper and many of my blog posts

**Review by ryzhyk — APPROVED** [review]
I think this is ok as a local fix. We'll need a way to educate people about these concepts in a more top-down fashion.

**Comment by mihaibudiu** [comment]
Blog posts will be forthcoming.
Maybe we don't need this construct at all?

**Comment by lalithsuresh** [comment]
Need these concepts in the docs too.

**Comment by ryzhyk** [comment]
> Need these concepts in the docs too.

Yes, that's what I meant. But that's a separate project.

**Pull request merged**